e2e/kind.sh

```diff
@@ -189,6 +200,15 @@ test_reject_peer_empty_public_key() {
 	assert_fail "create_peer e2e 10.5.0.1/32 0 ''" "should not be able to create Peer with empty public key"
 }
 
+test_fullmesh_allowed_location_ips() {
+	docker exec kind-cluster-kilo-control-plane ip address add 10.6.0.1/32 dev eth0
+	$KUBECTL_BINARY annotate node kind-cluster-kilo-control-plane kilo.squat.ai/allowed-location-ips=10.6.0.1/32
+	assert_equals Unauthorized "$(retry 10 5 'IP is not yet routable' curl_pod -m 1 -s -k https://10.6.0.1:10250/healthz)" "should be able to make HTTP request to allowed location IP"
+	$KUBECTL_BINARY annotate node kind-cluster-kilo-control-plane kilo.squat.ai/allowed-location-ips-
+	assert "retry 10 5 'IP is still routable' _not curl_pod -m 1 -s -k https://10.6.0.1:10250/healthz" "should not be able to make HTTP request to allowed location IP"
+	docker exec kind-cluster-kilo-control-plane ip address delete 10.6.0.1/32 dev eth0
```

Comment:
Did unit bash provide assert_fail or something?

Reply:
yes it does, however we need to be able to retry the `curl_pod` function, as the cluster might still be reconciling after removing the annotation. In order to be able to use the `retry` function, we need to invert the exit code from the `curl_pod` function. If we want to use `assert_fail` then we'd need to also invert the exit code of the `retry` function also, like` assert "_not retry 10 5 '' _not curl_pod ...`